example/zipIterator/src/zipIterator-main.cpp

```diff
@@ -0,0 +1,245 @@
+/* Copyright 2021 Hauke Mewes, Simeon Ehrig, Victor
+ *
+ * This file is part of vikunja.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#include <vikunja/transform/transform.hpp>
+#include <vikunja/mem/iterator/ZipIterator.hpp>
+
+#include <alpaka/alpaka.hpp>
+
+#include <iostream>
+#include <variant>
+
+template<std::size_t I = 0, typename FuncT, typename... Tp>
+inline typename std::enable_if<I == sizeof...(Tp), void>::type forEach(std::tuple<Tp...> &, FuncT) // Unused arguments are given no names
+{
+}
+
+template<std::size_t I = 0, typename FuncT, typename... Tp>
+inline typename std::enable_if<I < sizeof...(Tp), void>::type forEach(std::tuple<Tp...>& t, FuncT f)
+{
+    f(std::get<I>(t));
+    forEach<I + 1, FuncT, Tp...>(t, f);
+}
+
+template<typename IteratorTuplePtr>
+void printTuple(IteratorTuplePtr tuple)
+{
+    std::cout << "tuple(";
+    int index = 0;
+    int tupleSize = std::tuple_size<IteratorTuplePtr>{};
+    forEach(tuple, [&index, tupleSize](auto &x) { std::cout << x << (++index < tupleSize ? ", " : ""); });
+    std::cout << ")";
+}
+
+int main()
+{
+    // Define the accelerator here. Must be one of the enabled accelerators.
+    using TAcc = alpaka::AccCpuSerial<alpaka::DimInt<3u>, std::uint64_t>;
+
+    // Types of the data that will be reduced
+    using TRed = uint64_t;
```

Comment:
`TRed`, `TRedChar` and `TRedDouble` is not useful in your example. Your example is to specialized. Therefore you can direct use the concrete types.